mr/src/main/java/org/elasticsearch/hadoop/serialization/bulk/ScriptTemplateBulk.java

```diff
@@ -38,7 +39,9 @@
     @Override
     protected void doWriteObject(Object object, BytesArray storage, ValueWriter<?> writer) {
         if (ConfigurationOptions.ES_OPERATION_UPSERT.equals(settings.getOperation())) {
-            super.doWriteObject(object, storage, writer);
+            if (settings.hasScriptUpsert()) {
+                super.doWriteObject(Collections.emptyMap(), storage, writer);
```

Comment:
DataFrameValueWriter::write can't handle this being a Collection rather than a Tuple2. But if I change this line to
```
storage.add("{}");
```
then it seems to work (at least for my one test case). There might be a cleaner way to do that.

Reply:
Maybe something like this instead:
```
                FastByteArrayOutputStream bos = new FastByteArrayOutputStream(storage);
                JacksonJsonGenerator generator = new JacksonJsonGenerator(bos);
                generator.writeBeginObject();
                generator.writeEndObject();
                generator.close();
```